src/app/profiles/store/reducers/popin.reducer.ts

```diff
@@ -0,0 +1,30 @@
+import { SET_POPIN, PopinActionReturn } from '../actions/popin';
+import { MouseEvent } from 'react';
+
+export interface ContextPopinState {
+  content: {
+    text: string;
+    btnLabel?: string;
+    onClick?: (params?: MouseEvent<HTMLButtonElement>) => void;
```

Comment:
It seem this onclick will alway apply to the btn.. mb your state would be better shaped smthing like:
```ts
{
    text: string; // or content
    btn?: ButtonDescriptor;
    opened: boolean;
}

interface ButtonDescriptor { 
      label?: string;
      onClick?: (params?: MouseEvent<HTMLButtonElement>) => void;
}
```

Your call.